src/timer.rs

```diff
@@ -281,10 +280,13 @@ macro_rules! hal {
         $(
             impl Timer<$TIMX> {
                 /// Initialize timer
-                pub fn $timX(tim: $TIMX, clocks: &Clocks, apb: &mut $APBx) -> Self {
+                pub fn $timX(tim: $TIMX, clocks: &Clocks) -> Self {
                     // enable and reset peripheral to a clean slate state
-                    $TIMX::enable(apb);
-                    $TIMX::reset(apb);
+                    unsafe {
+                        let rcc = &(*RCC::ptr());
+                        $TIMX::enable(rcc);
+                        $TIMX::reset(rcc);
```

Comment:
This note applies to all copies of `enable` and `reset` function calls: Should we not move them out of the unsafe block? They are safe to call so they don't need to be there as far as I can tell?

On the other hand, does it make sense to not have them be unsafe. We could make them public API and mark them as unsafe in case people want to circumvent part of the HAL in a project but still want to init the peripherals via RCC

Reply:
> This note applies to all copies of `enable` and `reset` function calls: Should we not move them out of the unsafe block? They are safe to call so they don't need to be there as far as I can tell?

Moved.

Reply:
> On the other hand, does it make sense to not have them be unsafe. We could make them public API and mark them as unsafe in case people want to circumvent part of the HAL in a project but still want to init the peripherals via RCC

We can, but I don't see sense to make them unsafe as `rcc` can be gotten in safe way only during initialization. In all other places we need to use unsafe `RCC::ptr()`.